Vungle/VungleAdapter/ALVungleMediationAdapter.m

```diff
@@ -205,21 +215,23 @@ - (void)loadInterstitialAdForParameters:(id<MAAdapterResponseParameters>)paramet
     NSString *placementIdentifier = parameters.thirdPartyAdPlacementIdentifier;
     [self log: @"Loading %@interstitial ad for placement: %@...", ( isBiddingAd ? @"bidding " : @"" ), placementIdentifier];
     
-    if ( ![VungleAds isInitialized] )
-    {
-        [self log: @"Vungle SDK not successfully initialized: failing interstitial ad load..."];
-        [delegate didFailToLoadInterstitialAdWithError: MAAdapterError.notInitialized];
-        
-        return;
-    }
-    
-    [self updateUserPrivacySettingsForParameters: parameters];
-    
-    self.interstitialAd = [[VungleInterstitial alloc] initWithPlacementId: placementIdentifier];
-    self.interstitialAdDelegate = [[ALVungleMediationAdapterInterstitialAdDelegate alloc] initWithParentAdapter: self andNotify: delegate];
-    self.interstitialAd.delegate = self.interstitialAdDelegate;
-    
-    [self.interstitialAd load: bidResponse];
+    [self initVungleSDK:(id<MAAdapterInitializationParameters>)parameters completionHandler:^(MAAdapterInitializationStatus initializationStatus, NSString * _Nullable errorMessage) {
```

Comment:
Is the protocol `MAAdapterResponseParameters` comparable to `MAAdapterInitializationParameters`?

Reply:
Yes the base protocol is the same.

Reply:
If you are only going to use something in the base protocol, then the method should require a `MAAdapterParameters` instead of `MAAdapterInitializationParameters` and these method calls to the `initVungleSDK` should be casting the `MAAdapterResponseParameters` to `MAAdapterParameters`.

That is safer because if the `MAAdapterInitializationParameters` gets a property added by MAX and it get accessed in the `initVungleSDK`, then it would crash if the developer doesn't didn't know it got casted to a different thing.